packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -1,31 +1,27 @@
 import * as React from 'react';
 import { View } from 'react-native';
-import { ShadowProps } from './Shadow.types';
+import { ShadowProps, ShadowTokens, shadowName } from './Shadow.types';
+import { compressible, buildUseTokens, useFluentTheme, mergeProps } from '@fluentui-react-native/framework';
+import { shadowStyleFromTheme } from './shadowStyle';
 
-export const Shadow: React.FunctionComponent<ShadowProps> = (props: ShadowProps) => {
-  const noOffset = { width: 0, height: 0 };
-  const neutralShadowAmbient = { shadowOffset: noOffset, shadowOpacity: 0.12, shadowRadius: 2 };
-  const neutralShadowAmbientDarker = { shadowOffset: noOffset, shadowOpacity: 0.2, shadowRadius: 8 };
-  const ambientShadow = {
-    '2': neutralShadowAmbient,
-    '4': neutralShadowAmbient,
-    '8': neutralShadowAmbient,
-    '16': neutralShadowAmbient,
-    '28': neutralShadowAmbientDarker,
-    '64': neutralShadowAmbientDarker,
-  };
-  const keyShadow = {
-    '2': { shadowOffset: { width: 0, height: 1 }, shadowOpacity: 0.14, shadowRadius: 2 },
-    '4': { shadowOffset: { width: 0, height: 2 }, shadowOpacity: 0.14, shadowRadius: 4 },
-    '8': { shadowOffset: { width: 0, height: 4 }, shadowOpacity: 0.14, shadowRadius: 8 },
-    '16': { shadowOffset: { width: 0, height: 8 }, shadowOpacity: 0.14, shadowRadius: 16 },
-    '28': { shadowOffset: { width: 0, height: 14 }, shadowOpacity: 0.24, shadowRadius: 28 },
-    '64': { shadowOffset: { width: 0, height: 32 }, shadowOpacity: 0.24, shadowRadius: 64 },
+const useShadowTokens = buildUseTokens<ShadowTokens>(() => ({}), shadowName);
+
+export const Shadow = compressible<ShadowProps, ShadowTokens>((props: ShadowProps) => {
+  const { depth } = props;
+  const theme = useFluentTheme();
+
+  const shadowStyle = shadowStyleFromTheme(theme, depth);
```

Comment:
Hm we probably want some sort of memoization here. Otherwise the style object gets recreated on every rerender and that would cause excessive rerenderings

Reply:
I'm assuming I can add the memoization in Shadow.tsx, with the cache that I see in other components that implement compressible? Let me know if you were thinking of something different

Reply:
Yeah, that should be able to work. You can use something similar to what other tokens do in other compressible components or you can use memoize